src/ZeroC.Slice/SliceEncoder.cs

```diff
@@ -354,7 +354,7 @@ public void EncodeVarUInt62(ulong v)
         int tag,
         TagFormat tagFormat,
         T v,
-        EncodeAction<T> encodeAction) where T : notnull
+        EncodeAction<T> encodeAction)
```

Comment:
Or maybe keep the `where T : notnull`  + `EncodeAction<T?>`?

Reply:
The problem with using `EncodeAction<T?>` is that we will have to add `!` 

```
encoder.EncodeTagged(
    20,
    TagFormat.OptimizedVSize,
    b,
    (ref SliceEncoder encoder, string value) => encoder.EncodeString(value));
````

Will become

```
encoder.EncodeTagged(
    20,
    TagFormat.OptimizedVSize,
    b,
    (ref SliceEncoder encoder, string? value) => encoder.EncodeString(value!));
```

Reply:
Do we need to allow encoding `null` proxy with EncodeTagged, for decoding we don't have any other option as Ice encodes it. I was thinking for encoding to don't allow distinguish between null and not set.

Reply:
See #3930. 
As I proposed on the issue, we should provide 2 separate fixes.

Reply:
> The problem with using EncodeAction<T?> is that we will have to add !

You're right, this is a dumb idea. And since this issue is only for custom types, we should not change the EncodeTagged API which is correct as it is.

Reply:
I kept the bounds on the function intact thanks to an idea by @bernardnormier.